support out-of-tree SPIRV-Tools and SPIRV-Headers where the spirv-dis target doesn't exist yet

For example, 

With this kind of command line:
```
cmake -GNinja -DCMAKE_BUILD_TYPE=Release -DSPIRV_TOOLS_SOURCE_DIR=/foosrc/third_party/SPIRV-Tools -DSPIRV_HEADERS_SOURCE_DIR=/foosrc/third_party/SPIRV-Headers ../clspv
```

Currently, I get an error like this:

```
CMake Error at test/CMakeLists.txt:17 (add_custom_target):
  Error evaluating generator expression:

    $<TARGET_FILE_DIR:spirv-dis>

  No target "spirv-dis"
```

In this case we then add the SPIRV-Tools project from that other location?
See comment on #202

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -50,16 +50,17 @@ if (NOT DEFINED SPIRV_HEADERS_SOURCE_DIR)
 endif()
 
 if (NOT DEFINED SPIRV_TOOLS_SOURCE_DIR)
-
   set(SPIRV_TOOLS_SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/third_party/SPIRV-Tools)
-
   use_component(${SPIRV_TOOLS_SOURCE_DIR})
+endif()
 
+if (NOT TARGET spirv-dis)
   # First tell SPIR-V Tools where to find SPIR-V Headers
   set(SPIRV-Headers_SOURCE_DIR ${SPIRV_HEADERS_SOURCE_DIR})
 
   # Bring in the SPIR-V Tools repository which we'll use for testing
-  add_subdirectory(${SPIRV_TOOLS_SOURCE_DIR} EXCLUDE_FROM_ALL)
+  add_subdirectory(${SPIRV_TOOLS_SOURCE_DIR}
+      ${CMAKE_CURRENT_BINARY_DIR}/third_party/SPIRV-Tools EXCLUDE_FROM_ALL)
 endif()
 
 set(SPIRV_TOOLS_BINARY_DIR "$<TARGET_FILE_DIR:spirv-dis>")

